src/lib/translators/toolchain.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Translation toolchain utilities.
+ *
+ * The main exported function is `getTranslator` that takes a
+ * backend name as input and returns the corresponding translator.
+ *
+ */
+import { PipelineStep } from '@/lib/steps';
+import { translators as mongoTranslators } from '@/lib/translators/mongo';
+
+type TransformerFunc = (pipeline: Array<PipelineStep>) => any;
+export type TranslatorRegistryType = { [backend: string]: TransformerFunc };
+
+const TRANSLATORS: TranslatorRegistryType = {};
+
+// register mongo translators
+for (const [backend, transformer] of Object.entries(mongoTranslators)) {
+  TRANSLATORS[backend] = transformer;
+}
+
+/** return a function translating an array of pipeline steps to expected backend
+ * steps
+ *
+ * Example usage:
+ * ```typescript
+ * const translator = getTranslator('mongo36');
+ * const result = translator(myPipeline);
+ * ```
+ */
+export function getTranslator(backend: string) {
```

Comment:
I was wondering why was this function better than just exporting the `TRANSLATORS` dictionary.
Is it to also have some safety at runtime?

I feel we either have to choose between being protected at compile time or at runtime.
If we go for the second option, I would have ditched this and ask users to rely on the first one.
Otherwise, we'll have to constantly support two error handling systems, and I fear we'll eventually forget to handle the second one at some point

Reply:
> I was wondering why was this function better than just exporting the `TRANSLATORS` dictionary.
> Is it to also have some safety at runtime?

I wanted to define an API rather than a raw object (which admittedly is also an API). This seemed like a reasonable sustainable encapsulation. If at some point we need to perform additional checks, the function will allow this more easily